### PR TITLE
Add support for i2c in imuBosh. 

### PR DESCRIPTION
**Pull request by Nicogene**
This PR add support for i2c in `imuBosh_BNO055`

In order to use it you need to install 
`$ sudo apt install libi2c-dev`
and in particular the `pc104` needs some custom kernel module to be compiled and loaded (`kempld-core.ko` and  `i2c-kempld.ko`)
Please review code.

### CODE REVIEW

**Comment by Nicogene** [comment]
Note that this device is not compiled in travis and in appveyor

File: cmake/FindI2C.cmake [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  mark_as_advanced(I2C_i2c_dev_h_INCLUDE_DIR)
+
+  file(READ "${I2C_i2c_dev_h_INCLUDE_DIR}/linux/i2c-dev.h" _i2c_dev_content)
+  if(NOT "${_i2c_dev_content}" MATCHES "i2c_smbus_access")
+    set(I2C_i2c_dev_h_INCLUDE_DIR I2C_i2c_dev_h_INCLUDE_DIR-NOTFOUND CACHE)
+  endif()
+
+  set(I2C_LIBRARIES "")
+  set(I2C_INCLUDE_DIRS ${I2C_i2c_dev_h_INCLUDE_DIR})
+  set(I2C_DEFINITIONS "")
```
Comment (traversaro) [inline cmake/FindI2C.cmake]: If `I2C_LIBRARIES` and `I2C_DEFINITIONS` are not necessary, I think we can remove them.

Comment (traversaro) [reply cmake/FindI2C.cmake]: Sorry, I just saw the rest of the code and I got it.

File: cmake/FindI2C.cmake [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                                    FOUND_VAR I2C_FOUND
+                                    REQUIRED_VARS I2C_INCLUDE_DIRS
+                                                  I2C_LIBRARIES)
+else()
+  find_path(I2C_i2c_dev_h_INCLUDE_DIR
+            NAMES linux/i2c-dev.h)
+  mark_as_advanced(I2C_i2c_dev_h_INCLUDE_DIR)
+
+  file(READ "${I2C_i2c_dev_h_INCLUDE_DIR}/linux/i2c-dev.h" _i2c_dev_content)
+  if(NOT "${_i2c_dev_content}" MATCHES "i2c_smbus_access")
```
Comment (traversaro) [inline cmake/FindI2C.cmake]: Do you have a link or reference to explain what is going on here?

Comment (Nicogene) [reply cmake/FindI2C.cmake]: In all linux machines exists the file `usr/include/i2c-dev.h` almost empty.
Installing `libi2c-dev` version < 3 will rename `usr/include/i2c-dev.h` in `usr/include/i2c-dev.h.kernel` and will install `usr/include/i2c-dev.h` with all the function declaration and definitions, so no library is installed.
For `libi2c-dev` > 4.0  it install `usr/include/i2c/smbus.h` and the library `i2c.a` and `i2c.so`
It is quite a mess, but I think we covered all the possible configurations :sweat_smile:

**Pull request merged**